sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1102,13 +1097,56 @@ def get_cmis_dp_init_duration_secs(self, api):
     def get_cmis_dp_deinit_duration_secs(self, api):
         return api.get_datapath_deinit_duration()/1000
 
-    def is_cmis_application_update_required(self, api, channel, speed):
+    def get_cmis_host_lane_mask(self, api, host_lane_count, channel, speed):
+        """
+        Retrieves mask of active host lanes based on host lane count, channel and speed
+
+        Args:
+            api:
+                XcvrApi object
+            host_lane_count:
+                Integer, number of lanes on the host side
+            channel:
+                Integer, channel id of the group which the host lanes belong to (1 based)
+            speed:
+                Integer, the port speed of the host interface
+
+        Returns:
+            Integer, a mask of the active lanes on the host side
+            e.g. 0x5 for lane 0 and lane 2.
+        """
+        host_lane_mask = 0
+
+        if host_lane_count <= 0 or channel < 0 or speed <= 0:
+            self.log_error("Invalid input to get host lane mask - host_lane_count {} channel {} speed {}!".format(
+                            host_lane_count, channel, speed))
+            return host_lane_mask
+
+        if channel != 0:
+            appl = self.get_cmis_application_desired(api, host_lane_count, speed)
+            if appl < 1:
+                self.log_error("Failed to get host lanes as no suitable app {} found host_lane_count {} speed {}".format(
+                                appl, host_lane_count, speed))
+                return host_lane_mask
+
+            host_lane_assignment_option = api.get_host_lane_assignment_option(appl)
+            host_lane_start_bit = (host_lane_count * (channel - 1))
```

Comment:
just wanna confirm that channel is 0 based o r 1 based? I guess (0 - 1) will generate a negative value, is it expected?

Reply:
Channel is 0 based wherein 0 means the ports doesn't have a breakout module.
Non zero channel number will point to the logical group of lanes we would use.

In this case, (0 - 1) is not a valid test case since this piece of code gets executed if and only if channel != 0. Also, we have a check in this function to ensure channel is >= 0.